backend/otel/otel.go

```diff
@@ -236,79 +155,90 @@ func (o *Handler) HandleTrace(w http.ResponseWriter, r *http.Request) {
 
 	spans := req.Traces().ResourceSpans()
 	for i := 0; i < spans.Len(); i++ {
-		var projectID, sessionID, requestID, service, host string
-		var source modelInputs.LogSource
 		resource := spans.At(i).Resource()
-		resourceAttributes := resource.Attributes().AsRaw()
-		setHighlightAttributes(resourceAttributes, &projectID, &sessionID, &requestID, &source, &service, &host)
 		scopeScans := spans.At(i).ScopeSpans()
 		for j := 0; j < scopeScans.Len(); j++ {
 			spans := scopeScans.At(j).Spans()
 			for k := 0; k < spans.Len(); k++ {
 				span := spans.At(k)
-				spanAttributes := span.Attributes().AsRaw()
-				setHighlightAttributes(spanAttributes, &projectID, &sessionID, &requestID, &source, &service, &host)
 				events := span.Events()
 				for l := 0; l < events.Len(); l++ {
 					event := events.At(l)
 					eventAttributes := event.Attributes().AsRaw()
-					setHighlightAttributes(eventAttributes, &projectID, &sessionID, &requestID, &source, &service, &host)
+
+					fields, err := extractFields(ctx, extractFieldsParams{
```

Comment:
Just to confirm - this method is completing the logic we used to complete down the line in `setHighlightAttributes`?